lib/collection/response.js

```diff
@@ -142,17 +142,24 @@ var util = require('../util'),
     supportsBuffer = (typeof Buffer !== undefined) && _.isFunction(Buffer.byteLength),
 
     /**
-     * Normalises an input Buffer or buffer.toJSON() into a Buffer or ArrayBuffer.
+     * Normalizes an input Buffer, Buffer.toJSON() or base64 string into a Buffer or ArrayBuffer.
      *
      * @private
-     * @param {Buffer|Object} stream - An instance of Buffer of an object representation of Buffer(Buffer.toJSON())
+     * @param {Buffer|Object|String} stream - An instance of Buffer, Buffer.toJSON(), or Buffer.toString('base64')
      * @returns {Buffer|ArrayBuffer|undefined}
      */
-    normaliseStream = function (stream) {
+    normalizeStream = function (stream) {
+        // create buffer from buffer's JSON representation
         if (stream && stream.type === BUFFER && _.isArray(stream.data)) {
             // @todo Add tests for Browser environments, where ArrayBuffer is returned instead of Buffer
-            return typeof Buffer === FUNCTION ? new Buffer(stream.data) : new Uint8Array(stream.data).buffer;
+            return typeof Buffer === FUNCTION ? Buffer.from(stream.data) : new Uint8Array(stream.data).buffer;
         }
+        // create buffer from base64 string
+        else if (typeof stream === 'string') {
+            return Buffer.from(stream, 'base64');
```

Comment:
Default stream encoding should be put up as a constant.

Reply:
What got in the way of doing dataURI equivalent?

Reply:
The data URI scheme (`data:[<media type>][;base64],<data>`) comes with the overhead of string manipulation for media type, whitespace inconsistency between `[;base64],` and `<data>` etc.

The best way to handle all that is by using RegEx which will hit performance for large response payload. And, cause a temporary memory consumption spike for the additional strings constructed.

I would rather prefer an object representation instead of embracing the limitations of the HTML attribute (as data URI is built for that).
```js
{
    encoding: "base64",
    data: "ZGF0YQ==",
    mimeInfo: {}
}
```

Reply:
The current approach enforces stream property to be either a valid:
- Buffer
- Buffer.toJSON()
- Buffer.toString('base64')